filebeat/crawler/crawler.go

```diff
@@ -91,7 +102,14 @@ func (c *Crawler) Stop() {
 		c.wg.Add(1)
 		go stopProspector(p)
 	}
+
+	if c.reloader != nil {
+		c.wg.Add(1)
+		go c.reloader.Stop()
```

Comment:
oh... I'm not really ok with calling it on line 50. When using waitGroups one must/should ensure `Add` is always run before `Done`, so the internal semaphore(counter) can not become negative at any time. What if `Run` returns early (even if it doesn't). With `Done` being called on line 50, let's move the add right before that line.